x-pack/plugins/security_solution/server/endpoint/routes/policy/handlers.test.ts

```diff
@@ -24,71 +25,194 @@ import { SearchResponse } from 'elasticsearch';
+      const response = createSearchResponse(new EndpointDocGenerator().generatePolicyResponse());
+      const hostPolicyResponseHandler = getHostPolicyResponseHandler({
+        logFactory: loggingSystemMock.create(),
+        service: endpointAppContextService,
+        config: () => Promise.resolve(createMockConfig()),
+      });
 
-    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() => Promise.resolve(response));
-    const mockRequest = httpServerMock.createKibanaRequest({
-      params: { agentId: 'id' },
+      mockScopedClient.callAsCurrentUser.mockImplementationOnce(() => Promise.resolve(response));
+      const mockRequest = httpServerMock.createKibanaRequest({
+        params: { agentId: 'id' },
+      });
+
+      await hostPolicyResponseHandler(
+        createRouteHandlerContext(mockScopedClient, mockSavedObjectClient),
+        mockRequest,
+        mockResponse
+      );
+
+      expect(mockResponse.ok).toBeCalled();
+      const result = mockResponse.ok.mock.calls[0][0]?.body as GetHostPolicyResponse;
+      expect(result.policy_response.agent.id).toEqual(response.hits.hits[0]._source.agent.id);
     });
 
-    await hostPolicyResponseHandler(
-      createRouteHandlerContext(mockScopedClient, mockSavedObjectClient),
-      mockRequest,
-      mockResponse
-    );
+    it('should return not found when there is no response policy for host', async () => {
+      const hostPolicyResponseHandler = getHostPolicyResponseHandler({
+        logFactory: loggingSystemMock.create(),
+        service: endpointAppContextService,
+        config: () => Promise.resolve(createMockConfig()),
+      });
+
+      mockScopedClient.callAsCurrentUser.mockImplementationOnce(() =>
+        Promise.resolve(createSearchResponse())
+      );
 
-    expect(mockResponse.ok).toBeCalled();
-    const result = mockResponse.ok.mock.calls[0][0]?.body as GetHostPolicyResponse;
-    expect(result.policy_response.agent.id).toEqual(response.hits.hits[0]._source.agent.id);
+      const mockRequest = httpServerMock.createKibanaRequest({
+        params: { agentId: 'id' },
+      });
+
+      await hostPolicyResponseHandler(
+        createRouteHandlerContext(mockScopedClient, mockSavedObjectClient),
+        mockRequest,
+        mockResponse
+      );
+
+      expect(mockResponse.notFound).toBeCalled();
+      const message = mockResponse.notFound.mock.calls[0][0]?.body;
+      expect(message).toEqual('Policy Response Not Found');
+    });
   });
 
-  it('should return not found when there is no response policy for host', async () => {
-    const hostPolicyResponseHandler = getHostPolicyResponseHandler({
-      logFactory: loggingSystemMock.create(),
-      service: endpointAppContextService,
-      config: () => Promise.resolve(createMockConfig()),
+  describe('test agent policy summary handler', () => {
+    let mockAgentService: jest.Mocked<AgentService>;
+
+    const agentListResult = {
```

Comment:
may want to consider moving these rich objects to the `beforeEach` in order to avoid possible object mutation. Or - use the `Immutable<>` type to add TS checking